modules/core/src/main/java/org/apache/ignite/internal/util/CompletableFutures.java

```diff
@@ -82,4 +83,27 @@ public static <T> CompletableFuture<Set<T>> emptySetCompletedFuture() {
     public static <K, V> CompletableFuture<Map<K, V>> emptyMapCompletedFuture() {
         return (CompletableFuture<Map<K, V>>) (CompletableFuture<?>) EMPTY_MAP_COMPLETED_FUTURE;
     }
+
+    /**
+     * Returns a future that is completed when all provided futures complete (the behavior is identical to {@link CompletableFuture#allOf}).
+     *
+     * <p>If the future completes successfully, it aggregates the results of the source futures into a {@code List}.
+     *
+     * @param cfs Source futures.
+     * @param <T> Type of the result of the source futures.
+     * @return Future that completes with a list of results from the source futures.
+     */
+    @SafeVarargs
+    public static <T> CompletableFuture<List<T>> allOf(CompletableFuture<T>... cfs) {
```

Comment:
This particular name might easily be confused with `CompletableFuture.allOf`, even the class name is similar. Any ideas for a better name?

Reply:
Why is that a problem?

Reply:
Maybe it's not a problem, I expressed my opinion. For me personally that's an issue, because now I'd have to disassociate `allOf` with one specific thing, and start associating it with two sligthly different things. I'm fine if you leave the old name, it's not a blocker comment